packages/react/src/use-observable.ts

```diff
@@ -1,19 +1,36 @@
 import { Observable } from '@frp-ts/core'
-import { useEffect, useState } from 'react'
+import { useRef, useCallback } from 'react'
+import { useSyncExternalStore } from 'use-sync-external-store/shim'
 
-export function useObservable<Value>(source: Observable<Value>, initial: Value): Value {
-	const [state, setState] = useState(() => initial)
+interface Ref<Value> {
+	source: Observable<Value>
+	value: Value
+}
 
-	useEffect(() => {
-		const subscription = source.subscribe({
-			next: (value) => {
-				setState(() => value)
-			},
-		})
-		return () => {
-			subscription.unsubscribe()
+export function useObservable<Value>(source: Observable<Value>, initial: Value): Value {
+	// eslint-disable-next-line no-restricted-syntax
+	const ref = useRef<Ref<Value>>(undefined as unknown as Ref<Value>)
```

Comment:
Why not `useRef<Ref<Value> | undefined>`?

Reply:
Because we'll get useless checks, `ts` is not that smart yet

Reply:
I'd still opt for those checks instead of type case + suppressing eslint comment